### PR TITLE
Fixed crashes

### PR DESCRIPTION
**Pull request by mnewlive**
**Related Issue**  
https://github.com/saltedge/sca-authenticator-android/issues/258
https://github.com/saltedge/sca-authenticator-android/issues/274
https://github.com/saltedge/sca-authenticator-android/issues/275

### CODE REVIEW

File: authenticator_sdk/src/main/kotlin/com/saltedge/authenticator/sdk/tools/sign/SignatureTools.kt [hunk 1]
```diff
@@ -34,14 +34,19 @@ fun createSignatureHeader(
...
     val payload = "${requestMethod.toLowerCase(Locale.US)}|$requestUrl|$expiresAt|$requestBody"
-    return payload.toByteArray(StandardCharsets.UTF_8).signWith(privateKey)?.let {
-        Base64.encodeToString(it, Base64.NO_WRAP)
-    } ?: return ""
+    return runCatching {
+        return payload.toByteArray(StandardCharsets.UTF_8).signWith(privateKey)?.let {
+            Base64.encodeToString(it, Base64.NO_WRAP)
+        } ?: return ""
+    }.onFailure {
+        it.printStackTrace()
```
Comment (ConstantinKV) [inline authenticator_sdk/src/main/kotlin/com/saltedge/authenticator/sdk/tools/sign/SignatureTools.kt:42]: not print , log the error

File: app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt [hunk 2]
```diff
@@ -23,49 +23,39 @@ package com.saltedge.authenticator.tools
...
  * Create an error collector for the application not for DEBUG version
  *
  * @return crashlytics
  */
 fun createCrashlyticsKit() {
-    FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(!BuildConfig.DEBUG)
+    if (BuildConfig.DEBUG) {
+        Timber.plant(DebugTree())
+    } else {
+        FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(!BuildConfig.DEBUG)
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt:38]: move up, outside `if (BuildConfig.DEBUG)` condition

File: app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt [hunk 3]
```diff
@@ -23,49 +23,39 @@ package com.saltedge.authenticator.tools
...
-    if (BuildConfig.DEBUG && message.isNotEmpty()) {
-        var start = 0
-        while (start < message.length) {
-            var end = start + MAX_LOG_SIZE
-            if (end > message.length) end = message.length
-            Log.d(tag, message.substring(start, end).trim())
-            start += MAX_LOG_SIZE
+class CrashlyticsReportingTree : Timber.Tree() {
+
+    override fun log(priority: Int, tag: String?, message: String, throwable : Throwable?) {
```
Comment (ConstantinKV) [inline app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt:50]: remove extra space after `throwable`.  
`message` is optional (see http://jakewharton.github.io/timber/timber/log/Timber.Tree.html#log-int-java.lang.String-java.lang.String-java.lang.Throwable-)

File: authenticator_sdk/src/main/kotlin/com/saltedge/authenticator/sdk/web/SEFrameWebView.kt [hunk 4]
```diff
@@ -46,6 +47,14 @@ open class SEFrameWebView : WebView {
         setupView()
     }
 
+    override fun setOverScrollMode(mode: Int) {
+        runCatching {
+            super.setOverScrollMode(mode)
```
Comment (ConstantinKV) [inline authenticator_sdk/src/main/kotlin/com/saltedge/authenticator/sdk/web/SEFrameWebView.kt:52]: Please simulate an exception in SDK and prove that it has been logged by crashlytics.

File: app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt (hunk 3 above)
Comment (mnewlive) [reply app/src/main/kotlin/com/saltedge/authenticator/tools/LogTools.kt:50]: 2. https://github.com/JakeWharton/timber/commit/cf676130541ae9912b15fa66ee2eacee7220ce07 discussed in mattermost

**Pull request merged**